index.js

```diff
@@ -29,33 +29,38 @@ exports.RaveLevel = class RaveLevel extends ManyLevelGuest {
     this[kOptions] = { keyEncoding, valueEncoding }
     this[kConnect] = this[kConnect].bind(this)
     this[kDestroy] = this[kDestroy].bind(this)
+  }
 
-    this.open(this[kConnect])
+  _open(options, cb) {
+    super._open(options, (err) => {
+      this[kConnect](err, cb);
+    });
   }
 
-  [kConnect] () {
-    // Check database status every step of the way
-    if (this.status !== 'open') {
-      return
+  [kConnect] (err, cb) {
+    if (err) {
+      return cb(err);
     }
 
     // Attempt to connect to leader as follower
     const socket = net.connect(this[kSocketPath])
 
     // Track whether we succeeded to connect
     let connected = false
-    const onconnect = () => { connected = true }
+    const onconnect = () => {
+      connected = true
+      if (cb) {
+        cb();
+      }
+      cb = null; // Prevent any possible multiple callbacks onconnect
+    }
     socket.once('connect', onconnect)
 
     // Pass socket as the ref option so we don't hang the event loop.
     pipeline(socket, this.createRpcStream({ ref: socket }), socket, () => {
       // Disconnected. Cleanup events
       socket.removeListener('connect', onconnect)
 
-      if (this.status !== 'open') {
-        return
-      }
-
```

Comment:
Same here, it should check for `open` and `opening`, but don't remove this check.